Remove `unwrap` from transforms and add `ungroup` to more blocks (#50385)
WordPress/gutenberg: docs/reference-guides/block-api/block-transforms.md

-   **type** _(string)_: the value `shortcode`.
-   **tag** _(string|array)_: the shortcode tag or list of shortcode aliases this transform can work with.

> -   **transform** \_(function, optional): a callback that receives the shortcode attributes as the first argument and the [WPShortcodeMatch](/packages/shortcode/README.md#next) as the second. It should return a block object or an array of block objects. When this parameter is defined, it will take precedence over the `attributes` parameter.

Reviewer: ```suggestion
-   **transform** _(function, optional)_: a callback that receives the shortcode attributes as the first argument and the [WPShortcodeMatch](/packages/shortcode/README.md#next) as the second. It should return a block object or an array of block objects. When this parameter is defined, it will take precedence over the `attributes` parameter.
```

    ]
},
```

## `ungroup` blocks

> Via the optional `transforms` key of the block configuration, blocks can use the `ungroup` subkey to define which inner blocks of a grouping block can be extracted, replacing the block.

Reviewer: It's not just/necessarily inner blocks: attributes may be mined as well, as we see with the Quote block's `citation` attribute.

Author: I updated a bit the wording here: https://github.com/WordPress/gutenberg/pull/50385/commits/2edc576a6c574449ed797edd76f6d8f7fb3bb832.

> If a block has an `ungroup` transform, it is considered ungroupable, without the requirement of being the default grouping block. The UI used to ungroup a block with this API, is the same with the one used for the default grouping block. In order for the ugroup UI to render, we must have a single grouping block selected, which also contains some inner blocks.

Reviewer: ```suggestion
If a block has an `ungroup` transform, it is considered ungroupable, without the requirement of being the default grouping block. The UI used to ungroup a block with this API is the same as the one used for the default grouping block. In order for the Ungroup button to be displayed, we must have a single grouping block selected, which also contains some inner blocks.
```

Via the optional `transforms` key of the block configuration, blocks can use the `ungroup` subkey to define which inner blocks of a grouping block can be extracted, replacing the block.

> If a block has an `ungroup` transform, it is considered ungroupable, without the requirement of being the default grouping block. The UI used to ungroup a block with this API is the same as the one used for the default grouping block. In order for the Ungroup button to be displayed, we must have a single grouping block selected, which also contains some inner blocks.

Reviewer 2: > it is considered ungroupable

I think this phrase might be confusing for folks because "ungroupable" means "Incapable of being grouped together".

Reviewer: Agreed. Depending on the use in a sentence, some ideas: "able to be ungrouped", "ungroup-ready", "eligible for ungrouping"? 😅

Author: You're right. I went for `eligible for ungrouping`.
